File: 2145.count-the-hidden-sequences.py
#
# @lc app=leetcode id=2145 lang=python3
#
# [2145] Count the Hidden Sequences
#
from typing import List

# @lc code=start
class Solution:
    def numberOfArrays(self, differences: List[int], lower: int, upper: int) -> int:
        net = 0
        maxx = 0
        minn = 0
        for d in differences:
            net += d
            maxx = max(net, maxx)
            minn = min(net, minn)
        rangee = maxx - minn
        minMax = lower + rangee
        return 0 if upper - minMax + 1 <= 0 else upper - minMax + 1            
# @lc code=end

File: 2145.count-the-hidden-sequences.cpp
/*
 * @lc app=leetcode id=2145 lang=cpp
 *
 * [2145] Count the Hidden Sequences
 */
#include <bits/stdc++.h>

using namespace std;

// @lc code=start
class Solution {
public:
    int numberOfArrays(vector<int>& differences, int lower, int upper) {
        long net = 0, minn = 0, maxx = 0;
        for (int d: differences)
        {
            net += d;
            minn = min(minn, net);
            maxx = max(net, maxx);
        }

        int range = maxx - minn;
        int numArrays = upper - lower + 1 - range;
        return max(numArrays, 0);
    }
};
// @lc code=end

